label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleValidatorImpl.java

```diff
@@ -123,31 +122,55 @@ private void validateJsonCondition(JsonCondition jsonCondition) {
     }
   }
 
+  private void validateKeyStringCondition(StringCondition stringCondition) {
+    validateNonDefaultPresenceOrThrow(stringCondition, StringCondition.OPERATOR_FIELD_NUMBER);
+    switch (stringCondition.getOperator()) {
+      case OPERATOR_EQUALS:
+      case OPERATOR_MATCHES_REGEX:
+        return;
+      default:
+        throwInvalidArgumentException(
+            String.format(
+                "Invalid operator for key condition: %s: %s",
+                getName(stringCondition), printMessage(stringCondition)));
+    }
+  }
+
   private void validateStringCondition(StringCondition stringCondition) {
     validateNonDefaultPresenceOrThrow(stringCondition, StringCondition.OPERATOR_FIELD_NUMBER);
-    if (stringCondition.getOperator().equals(OPERATOR_MATCHES_IPS)
-        || stringCondition.getOperator().equals(OPERATOR_NOT_MATCHES_IPS)) {
-      switch (stringCondition.getKindCase()) {
-        case VALUE:
-          final String ip = stringCondition.getValue();
-          if (!isValidIpAddressOrSubnet(ip)) {
-            throwInvalidArgumentException(
-                "StringCondition in LabelApplicationRule should have a valid IP address or CIDR");
-          }
-          break;
-        case VALUES:
-          if (stringCondition.getValues().getValuesList().stream()
-              .anyMatch(s -> !isValidIpAddressOrSubnet(s))) {
+    switch (stringCondition.getOperator()) {
+      case OPERATOR_MATCHES_REGEX:
+        final String pattern = stringCondition.getValue();
+        Status regexValidationStatus = RegexValidator.validate(pattern);
```

Comment:
Does an empty string pass validation here (it shouldn't)?

Reply:
added validations